### PR TITLE
Log warning about TLS-SNI deprecation in Certbot

### PR DESCRIPTION
**Pull request by ohemorange**
Be sure to edit the `master` section of `CHANGELOG.md` with a line describing this PR before it gets merged.

For  #6319.

### CODE REVIEW

**Comment by bmw** [comment]
I was initially thinking of something like https://github.com/certbot/certbot/commit/20bcb1181094cf127e4fb4a73d0a15b1637c9242. We could instead do it in [challb_to_achall](https://github.com/certbot/certbot/blob/v0.27.1/certbot/auth_handler.py#L358) where it logs each challenge we're doing, but I'm not sure if we want to log it for every challenge rather than just the first occurrence of it.

What do you think? Do you prefer your approach?

**Comment by ohemorange** [comment]
I really don't think it matters either way.

File: certbot-apache/certbot_apache/configurator.py [hunk 1]
```diff
@@ -2276,6 +2276,9 @@ def perform(self, achalls):
                 http_doer.add_chall(achall, i)
             else:  # tls-sni-01
                 sni_doer.add_chall(achall, i)
+                logger.warning("TLS-SNI-01 is deprecated, and will stop working in February 2019. "
```
Comment (bmw) [inline certbot-apache/certbot_apache/configurator.py]: I don't have a strong preference, but I'd personally like us to remove the date here as things like that can change but a released version of Certbot cannot.

**Comment by ohemorange** [comment]
I'm seeing that we do check if `pref_challs` is set before using it, where are you talking about?

**Pull request merged**